### PR TITLE
Upgrade kubernetes dashboard from v2.3.1 to v2.5.1

### PR DESCRIPTION
**Pull request by ziyi-xie**
Kubernetes dashboard has been upgraded to v2.5.1 now, and this PR fixed the version of dashboard to v2.5.1 in minikube.
https://github.com/kubernetes/dashboard/releases

### CODE REVIEW

File: pkg/minikube/assets/addons.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ var Addons = map[string]*Addon{
 		MustBinAsset(addons.DashboardAssets, "dashboard/dashboard-secret.yaml", vmpath.GuestAddonsDir, "dashboard-secret.yaml", "0640"),
 		MustBinAsset(addons.DashboardAssets, "dashboard/dashboard-svc.yaml", vmpath.GuestAddonsDir, "dashboard-svc.yaml", "0640"),
 	}, false, "dashboard", "kubernetes", map[string]string{
-		"Dashboard":      "kubernetesui/dashboard:v2.3.1@sha256:ec27f462cf1946220f5a9ace416a84a57c18f98c777876a8054405d1428cc92e",
+		"Dashboard":      "kubernetesui/dashboard:v2.5.0@sha256:ec27f462cf1946220f5a9ace416a84a57c18f98c777876a8054405d1428cc92e",
```
Comment (ziyi-xie) [inline pkg/minikube/assets/addons.go:137]: Hi! I think I have to change the hash here, but the hash for v2.3.1 is different from the one I checked on dockerhub. Could you tell me where can I find it?

Comment (atoato88) [reply pkg/minikube/assets/addons.go:137]: You can get the digest(kubernetesui/dashboard:v2.3.1@sha256:ec27f462cf1946220f5a9ace416a84a57c18f98c777876a8054405d1428cc92e) by `docker inspect kubernetesui/dashboard:v2.3.1 --format='{{index .RepoDigests}}'`.
Note that you need to pull the image(`kubernetesui/dashboard:v2.3.1`) to local, in advance.

Comment (ziyi-xie) [reply pkg/minikube/assets/addons.go]: Thank you so much for your help!!

**Comment by atoato88** [comment]
Could you add some contexts of this PR on description?
- What this PR does / why we need it
- Which issue(s) this PR fixes (if exists)

File: pkg/minikube/cruntime/containerd_test.go [hunk 2]
```diff
@@ -27,7 +27,7 @@ func TestAddRepoTagToImageName(t *testing.T) {
 		imgName string
 		want    string
 	}{
-		{"kubernetesui/dashboard:v2.1.0", "docker.io/kubernetesui/dashboard:v2.1.0"},
+		{"kubernetesui/dashboard:v2.5.0", "docker.io/kubernetesui/dashboard:v2.5.0"},
 		{"kubernetesui/metrics-scraper:v1.0.4", "docker.io/kubernetesui/metrics-scraper:v1.0.4"},
```
Comment (t-inu) [inline pkg/minikube/cruntime/containerd_test.go:31]: In the previous upgrade, this file has not been updated, so metrics-scraper is not the latest version.
I think it's better to update from v1.0.4 to v1.0.7.

Comment (ziyi-xie) [reply pkg/minikube/cruntime/containerd_test.go]: It was corrected and thank you so much for letting me know!

**Comment by ziyi-xie** [comment]
@medyagh Hi! Thank you for running the test for me! And I noticed that there are several tests failed and I don't have any clue about the reason. Could you tell me what I can do to fix it? Looking forward to hearing from you and many thanks. :)

**Comment by shu-mutou** [comment]
@ziyi-xie Dashboard 2.5.1 was released last night. Could you bump to it?

**Review by klaases — APPROVED** [review]
Do you want to bump to v2.5.1?

**Comment by ziyi-xie** [comment]
> @ziyi-xie [Dashboard 2.5.1](https://github.com/kubernetes/dashboard/releases/tag/v2.5.1) was released last night. Could you bump to it?

It was corrected and thank you so much for letting me know!

**Comment by ziyi-xie** [comment]
> Do you want to bump to v2.5.1?

@klaases I've just bumped it to v2.5.1 and could you check if everything looks OK please? Thank you.

File: pkg/minikube/assets/addons.go [hunk 3]
```diff
@@ -134,7 +134,7 @@ var Addons = map[string]*Addon{
 		MustBinAsset(addons.DashboardAssets, "dashboard/dashboard-secret.yaml", vmpath.GuestAddonsDir, "dashboard-secret.yaml", "0640"),
 		MustBinAsset(addons.DashboardAssets, "dashboard/dashboard-svc.yaml", vmpath.GuestAddonsDir, "dashboard-svc.yaml", "0640"),
 	}, false, "dashboard", "kubernetes", map[string]string{
-		"Dashboard":      "kubernetesui/dashboard:v2.3.1@sha256:ec27f462cf1946220f5a9ace416a84a57c18f98c777876a8054405d1428cc92e",
+		"Dashboard":      "kubernetesui/dashboard:v2.5.1@sha256:cc746e7a0b1eec0db01cbabbb6386b23d7af97e79fa9e36bb883a95b7eb96fe2]",
```
Comment (shu-mutou) [inline pkg/minikube/assets/addons.go:137]: Please remove it. Thanks for bump to 2.5.1!
But the `]` at the last of hash is not needed.

Comment (ziyi-xie) [reply pkg/minikube/assets/addons.go]: I've just corrected it and thank you so much!

**Comment by ziyi-xie** [comment]
@spowelljr Hi! Sorry for the sudden message. Would it be possible for you to check if everything is ok with this PR for me please? Many thanks.

**Review by spowelljr — APPROVED** [review]

File: pkg/minikube/bootstrapper/images/kubeadm_test.go [hunk 4]
```diff
@@ -103,7 +103,7 @@ func TestKubeadmImages(t *testing.T) {
 			"k8s.gcr.io/etcd:3.2.24",
 			"k8s.gcr.io/pause:3.1",
 			"gcr.io/k8s-minikube/storage-provisioner:" + version.GetStorageProvisionerVersion(),
-			"docker.io/kubernetesui/dashboard:v2.3.1",
+			"docker.io/kubernetesui/dashboard:v2.5.1",
```
[the remark this answers is not in the document]
Comment (ziyi-xie) [reply pkg/minikube/bootstrapper/images/kubeadm_test.go:106]: Noted with many thanks.

**Comment by jesperpedersen** [comment]
This gives
```
❌  Unable to load cached images: loading cached images: stat /home/demo/.minikube/cache/images/amd64/docker.io/kubernetesui/dashboard_v2.5.1: no such file or directory
```

To see this:
```
minikube delete
minikube start
```

**Comment by ziyi-xie** [comment]
> This gives
> 
> ```
> ❌  Unable to load cached images: loading cached images: stat /home/demo/.minikube/cache/images/amd64/docker.io/kubernetesui/dashboard_v2.5.1: no such file or directory
> ```
> 
> To see this:
> 
> ```
> minikube delete
> minikube start
> ```

Thank you for letting me know!  But I didn't encounter such situation.
Is there any possibility that all the caches are cleared during the update, and will it be fixed if pull the image from dockerhub again before start the minikube?

**Pull request merged**